src/components/screens/on-call-calendar-screen.tsx

```diff
@@ -0,0 +1,259 @@
+import { useEffect, useState } from "react";
+import { DateCalendar, PickersDay, type PickersDayProps } from "@mui/x-date-pickers";
+import {
+  Badge,
+  Box,
+  CircularProgress,
+  FormControl,
+  InputLabel,
+  MenuItem,
+  Select,
+  Typography
+} from "@mui/material";
+import { DateTime } from "luxon";
+import { useLambdasApi } from "../../hooks/use-api";
+import { onCallAtom } from "../../atoms/oncall";
+import { useAtom, useSetAtom } from "jotai";
+import { errorAtom } from "../../atoms/error";
+import strings from "../../localization/strings";
+import { stringToColor } from "../../utils/oncall-utils";
+import type { OnCallCalendarEntry } from "../../types";
+import OnCallHandler from "../contexts/oncall-handler";
+import OnCallListView from "../oncall-calendars/oncall-list-view";
+import type { OnCallPaid } from "src/generated/homeLambdasClient";
+
+/**
+ * On call calendar screen component
+ */
+const OnCallCalendarScreen = () => {
+  /**
+   * Validates if local storage item is of correct type
+   * @returns boolean value from local storage if validated
+   */
+  const validateJSONString = () => {
+    const item = localStorage.getItem("calendarViewAsDefault");
+    if (item) return JSON.parse(item);
+    return true;
+  };
+
+  const { onCallApi } = useLambdasApi();
+  const [onCallData, setOnCallData] = useAtom(onCallAtom);
+  const [open, setOpen] = useState(false);
+  const [isCalendarView, setIsCalendarView] = useState(validateJSONString());
+  const [selectedDate, setSelectedDate] = useState<DateTime>(DateTime.now());
+  const [onCallPerson, setOnCallPerson] = useState("");
+  const [selectedOnCallEntry, setSelectedOnCallEntry] = useState<OnCallCalendarEntry>();
+  const setError = useSetAtom(errorAtom);
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    getOnCallData(selectedDate.year);
+  }, [selectedDate.year]);
+
+  // biome-ignore lint/correctness/useExhaustiveDependencies: dependency is necessary
+  useEffect(() => {
```

Comment:
I'm a bit suspicious of this, no need for changes now but keep this UE in mind as you start fixing bugs.